src/main/drivers/serial_uart_stm32f10x.c

```diff
@@ -89,200 +221,106 @@ void uart_tx_dma_IRQHandler(dmaChannelDescriptor_t* descriptor)
         s->txDMAEmpty = true;
 }
 
-#ifdef USE_UART1
-// USART1 - Telemetry (RX/TX by DMA)
-uartPort_t *serialUART1(uint32_t baudRate, portMode_t mode, portOptions_t options)
+uartPort_t *serialUART(UARTDevice device, uint32_t baudRate, portMode_t mode, portOptions_t options)
 {
-    uartPort_t *s;
-    static volatile uint8_t rx1Buffer[UART1_RX_BUFFER_SIZE];
-    static volatile uint8_t tx1Buffer[UART1_TX_BUFFER_SIZE];
+    uartDevice_t *uartDev = uartHardwareMap[device];
+    if (!uartDev)
+        return NULL;
+
+    uartPort_t *s = &(uartDev->port);
 
-    s = &uartPort1;
     s->port.vTable = uartVTable;
 
     s->port.baudRate = baudRate;
 
-    s->port.rxBuffer = rx1Buffer;
-    s->port.txBuffer = tx1Buffer;
-    s->port.rxBufferSize = UART1_RX_BUFFER_SIZE;
-    s->port.txBufferSize = UART1_TX_BUFFER_SIZE;
+    s->port.rxBuffer = uartDev->rxBuffer;
+    s->port.txBuffer = uartDev->txBuffer;
+    s->port.rxBufferSize = UART_RX_BUFFER_SIZE;
+    s->port.txBufferSize = UART_TX_BUFFER_SIZE;
 
-    s->USARTx = USART1;
+    s->USARTx = uartDev->dev;
 
-#ifdef USE_UART1_RX_DMA
-    dmaInit(DMA1_CH5_HANDLER, OWNER_SERIAL_RX, 1);
-    s->rxDMAChannel = DMA1_Channel5;
-    s->rxDMAPeripheralBaseAddr = (uint32_t)&s->USARTx->DR;
-#endif
-    s->txDMAChannel = DMA1_Channel4;
-    s->txDMAPeripheralBaseAddr = (uint32_t)&s->USARTx->DR;
+    if (uartDev->rxDMAChannel) {
+        dmaInit(dmaGetIdentifier(uartDev->rxDMAChannel), OWNER_SERIAL_RX, RESOURCE_INDEX(device));
+        s->rxDMAChannel = uartDev->rxDMAChannel;
+        s->rxDMAPeripheralBaseAddr = (uint32_t)&s->USARTx->DR;
+    }
+
+    RCC_ClockCmd(uartDev->rcc, ENABLE);
```

Comment:
IMO it's not necessary (but possible). I don't have reference, but there was some note in datasheet/errata that unclocked DMA source may cause problem.  So it is better to play it safe